diff --git pcmanfm2/src/desktop.c pcmanfm2/src/desktop.c
index d46e2c1..1a93641 100644
--- pcmanfm2/src/desktop.c
+++ pcmanfm2/src/desktop.c
@@ -178,7 +178,7 @@ static void fm_desktop_class_init(FmDesktopClass *klass)
     GtkObjectClass *gtk_object_class;
     GtkWidgetClass* wc;
     typedef gboolean (*DeleteEvtHandler) (GtkWidget*, GdkEvent*);
-    const char* atom_names[] = {"_NET_WORKAREA", "_NET_NUMBER_OF_DESKTOPS", "_NET_CURRENT_DESKTOP", "_XROOTMAP_ID"};
+    const char* atom_names[] = {"_NET_WORKAREA", "_NET_NUMBER_OF_DESKTOPS", "_NET_CURRENT_DESKTOP", "_XROOTPMAP_ID"};
     Atom atoms[G_N_ELEMENTS(atom_names)] = {0};
 
     /* g_object_class = G_OBJECT_CLASS(klass);
