diff --git a/pcmanfm2/src/main-win.c b/pcmanfm2/src/main-win.c
index b7a173d..2765a83 100644
--- a/src/main-win.c
+++ b/src/main-win.c
@@ -702,6 +702,7 @@ void on_new_tab(GtkAction* act, FmMainWin* win)
 void on_close_win(GtkAction* act, FmMainWin* win)
 {
     gtk_widget_destroy(GTK_WIDGET(win));
+    on_delete_event(win, NULL);
 }
 
 void on_close_tab(GtkAction* act, FmMainWin* win)
@@ -709,6 +710,7 @@ void on_close_tab(GtkAction* act, FmMainWin* win)
     GtkNotebook* nb = (GtkNotebook*)win->notebook;
     /* remove current page */
     gtk_notebook_remove_page(nb, gtk_notebook_get_current_page(nb));
+    on_delete_event(win, NULL);
 }
 
 void on_open_in_new_tab(GtkAction* act, FmMainWin* win)
